src/nvim/eval.c

```diff
@@ -5606,18 +5601,61 @@ int garbage_collect(void)
   for (int i = 1; i <= ga_scripts.ga_len; ++i)
     set_ref_in_ht(&SCRIPT_VARS(i), copyID);
 
-  /* buffer-local variables */
   FOR_ALL_BUFFERS(buf) {
+    // buffer-local variables
     set_ref_in_item(&buf->b_bufvar.di_tv, copyID);
+    // buffer marks (ShaDa additional data)
+    set_ref_in_fmark(buf->b_last_cursor, copyID);
+    set_ref_in_fmark(buf->b_last_insert, copyID);
+    set_ref_in_fmark(buf->b_last_change, copyID);
+    for (size_t i = 0; i < NMARKS; i++) {
+      set_ref_in_fmark(buf->b_namedm[i], copyID);
+    }
+    // buffer change list (ShaDa additional data)
+    for (int i = 0; i < buf->b_changelistlen; i++) {
+      set_ref_in_fmark(buf->b_changelist[i], copyID);
+    }
+    // buffer ShaDa additional data
+    set_ref_dict(buf->additional_data, copyID);
   }
 
-  /* window-local variables */
   FOR_ALL_TAB_WINDOWS(tp, wp) {
+    // window-local variables
     set_ref_in_item(&wp->w_winvar.di_tv, copyID);
+    // window jump list (ShaDa additional data)
+    for (int i = 0; i < wp->w_jumplistlen; i++) {
+      set_ref_in_fmark(wp->w_jumplist[i].fmark, copyID);
+    }
   }
   if (aucmd_win != NULL)
     set_ref_in_item(&aucmd_win->w_winvar.di_tv, copyID);
 
+  // registers (ShaDa additional data)
+  {
+    const void *reg_iter = NULL;
+    do {
+      yankreg_T reg;
+      char name = NUL;
```

Comment:
16.08.2015, 23:00, "oni-link" <notifications@github.com>:
> In src/nvim/eval.c:
>
>> } > if (aucmd_win != NULL) > set_ref_in_item(&aucmd_win->w_winvar.di_tv, copyID); > > + // registers (ShaDa additional data) > + { > + const void *reg_iter = NULL; > + do { > + yankreg_T reg; > + char name = NUL;
>
> op_register_iter(reg_iter, &name, &reg) also sets name to NUL.

I know. This is intentional, “just in case”.

> —
> Reply to this email directly or view it on GitHub.